syntax = "proto3";

package todoPackage;

service Todo {
  rpc createTodo(TodoItem) returns (TodoItem);
  rpc getTodo(RequestId) returns (TodoItem);
  rpc updateTodo(TodoItem) returns (TodoItem);
  rpc deleteTodo(RequestId) returns (Empty);
  rpc listTodos(Empty) returns (TodoItems);
  rpc save(Empty) returns (Empty);
}

message Empty {}

message RequestId {
  string id = 1;
}

message TodoItem {
  string id = 1;
  string text = 2;
}

message TodoItems {
  TodoItem items = 1;
}